pkg/obs/controllerserver.go

```diff
@@ -120,7 +142,35 @@ func (cs *controllerServer) ControllerUnpublishVolume(_ context.Context, _ *csi.
 
 func (cs *controllerServer) ListVolumes(_ context.Context, req *csi.ListVolumesRequest) (
 	*csi.ListVolumesResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "")
+	log.Infof("ListVolumes: called with args %v", protosanitizer.StripSecrets(*req))
+
+	if req.MaxEntries < 0 {
+		return nil, status.Error(codes.InvalidArgument,
+			fmt.Sprintf("Validation failed, max entries request %v, must not be negative ", req.MaxEntries))
+	}
+
+	volumes, err := services.ListBuckets(cs.Driver.cloud)
```

Comment:
Paginated queries should be used instead of querying all.

Reply:
done